src/dpq2/conv/from_d_types.d

```diff
@@ -75,4 +122,61 @@ unittest
         assert(t.as!bool == true);
         assert(f.as!bool == false);
     }
+
+    {
+        Value v = toValue(Nullable!long(1));
+        Value nv = toValue(Nullable!bool.init);
+
+        assert(!v.isNull);
+        assert(v.oidType == OidType.Int8);
+        assert(v.as!long == 1);
+
+        assert(nv.isNull);
+        assert(nv.oidType == OidType.Bool);
+    }
+
+    {
+        // Date: '2018-1-15' -> [0, 0, 25, 189]
+        auto d = Date(2018, 1, 15);
+        auto v = toValue(d);
+
+        assert(v.data == [0, 0, 25, 189]);
```

Comment:
This is the main comment throughout PR: I really do not like this magic.

More suitable is check by integration test:
1. create all of theese new Values
2. send it into Postgres server by "SELECT $1::text, $2::text, $3::text" (or use "exec". Here is only important what result should be a textual)
3. read answer from server
4. check values by concatination of expected string values. Postgres prints beautiful textual representations of datas, times etc.

Thus we make sure that we understand Postgres binary protocol correctly and the server correctly understands us.

integration test example:https://github.com/denizzzka/dpq2/blob/master/src/dpq2/conv/to_d_types.d#L178

This `_integration_test` functions called from: https://github.com/denizzzka/dpq2/blob/master/integration_tests/integration_tests.d

Reply:
I've removed the tests for "magic" byte arrays but kept the ones to test to/from Value conversions.
Then I've added the integration tests to the current ones - just extended existing ones to test all values the other way around.
There is one BUT and that is, that some value types don't work the other way around (there is a static if for them with a todo description).
As they didn't work event before, I would like to propose that we'll make a new issue for it and fix it in other PR to keep this one smaller?

Reply:
> As they didn't work event before, I would like to propose that we'll make a new issue for it and fix it in other PR to keep this one smaller?

Ok